internal/ccsmp/ccsmp_helper.c

```diff
@@ -54,103 +49,157 @@ solClientgo_msg_isRequestReponseMsg(solClient_opaqueMsg_pt msg_p, char **correla
 }
 
 solClient_returnCode_t  
-_SessionTopicSubscribeWithFlags( solClient_opaqueSession_pt opaqueSession_p,
-                                const char                *topicSubscription_p,
+SessionFlowCreate( solClient_opaqueSession_pt   opaqueSession_p,
+                    solClient_propertyArray_pt  flowPropsP,
+                    solClient_opaqueFlow_pt         *opaqueFlow_p,
+                    solClient_flow_createFuncInfo_t *funcInfo_p,
+                    solClient_uint64_t              flowID) 
+{
+    /* set the flowID in the flow create struct */
+    funcInfo_p->rxMsgInfo.user_p = (void *) flowID;
+	funcInfo_p->eventInfo.user_p = (void *) flowID;
+
+    return solClient_session_createFlow(flowPropsP, opaqueSession_p, opaqueFlow_p, funcInfo_p, sizeof(*funcInfo_p));
+}
```

Comment:
This is not safe. We're allocating funcInfo_p in GoLang but setting the size in C code.  I realize they are using the same header file and 'should' be correctly aligned but it's splitting up responsibility into two places which is bad coding practice. The 'sizeof' should be set by the code that allocates the structure. 

We need to put the 'unsafe.Sizeof()' argument back into the method signature, or pass all the structure members  separately and allocate the funcInfo in this code.

Reply:
This change is not addressed the original issue? As you still pass the flow create func info from Go runtime without the sizeof information from go. This does not allocate the func info in the native function but in go. And you still take the sizeof of the go runtime struct with all the same allignement concerns @RagnarPaulson mentioned.

Reply:
@cjwmorgan-sol, when I allocate the `funcInfo_p` in C (the ccsmp_helper.c) and run the integration tests I hit a bus error (segfault) because somehow the `funcInfo_p` or part of it is collected by the GC (C garbage collector, maybe ?) or pointing of unallocated/unusable memory at the time it is being used. 

Please do you know how to fix this (I was stuck on this for the most part of yesterday). I do see that we take this same approach in other part of the code, example here - `internal/ccsmp/ccsmp_core.go` line `276`